### PR TITLE
Fixed issue #19139: CSRF in Reset Survey menus (Tests)

### PR DESCRIPTION
**Pull request by gabrieljenik**
Tests for remote control: Real connection through RC client

### CODE REVIEW

File: tests/functional/backend/RemoteControlTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        $urlMan = Yii::app()->urlManager;
+        $urlMan->setBaseUrl(self::$tmpBaseUrl);
+
+        \SettingGlobal::setSetting('RPCInterface', self::$tmpRPCType);
+    }
+
+    public function testGetSessionKey()
+    {
+        $sessionKey = self::$client->call('get_session_key', ['admin', 'password']);
+        $this->assertIsNotArray($sessionKey);
```
Comment (Shnoulle) [inline tests/functional/backend/RemoteControlTest.php:58]: [assertIsString()](https://docs.phpunit.de/en/9.6/assertions.html#assertisstring) ?

A 500 error is not an array , right ?

**Review by Shnoulle — COMMENTED** [review]
Why not https://docs.phpunit.de/en/9.6/assertions.html#assertisstring ?

File: tests/functional/backend/RemoteControlTest.php (hunk 1 above)
Comment (gabrieljenik) [reply tests/functional/backend/RemoteControlTest.php:58]: I believe a 500 would be shown as an exception. Not sure.
Should try maybe...

Comment (gabrieljenik) [reply tests/functional/backend/RemoteControlTest.php:58]: Very edge case. Does it worth spending on it?

Comment (Shnoulle) [reply tests/functional/backend/RemoteControlTest.php:58]: > Very edge case. Does it worth spending on it?

The good result **MUST** be a string : then just check if it's a string.

Comment (gabrieljenik) [reply tests/functional/backend/RemoteControlTest.php:58]: This is what `get_session_key` returns.
```
        if (is_string($loginResult)) {
            return array('status' => $loginResult);
        }
        return array('status' => 'Invalid user name or password');
```

Comment (gabrieljenik) [reply tests/functional/backend/RemoteControlTest.php:58]: Also, jsonRpcClient, if something fails, throws Exception.
I would just leave it like this.

Comment (Shnoulle) [reply tests/functional/backend/RemoteControlTest.php:58]: > This is what `get_session_key` returns.
> 
> ```
>         if (is_string($loginResult)) {
>             return array('status' => $loginResult);
>         }
>         return array('status' => 'Invalid user name or password');
> ```

In case of error only 

In case of success: 
https://github.com/LimeSurvey/LimeSurvey/blob/5d97f49a06b610b32db927d434de76e88e1746a1/application/helpers/remotecontrol/remotecontrol_handle.php#L60


Else : if there are a CRSF issue : it's an Exception.

Comment (gabrieljenik) [reply tests/functional/backend/RemoteControlTest.php]: You are right

**Review by Shnoulle — APPROVED** [review]
PS : maybe check if it's an array with a bad password before ?

**Comment by gabrieljenik** [comment]
> PS : maybe check if it's an array with a bad password before ?

Added

**Comment by Shnoulle** [comment]
> > PS : maybe check if it's an array with a bad password before ?
> 
> Added

How, it's just an idea like this :)

Not mandatory

**Review by Shnoulle — APPROVED** [review]
